docs/DeveloperGuide.md

```diff
@@ -1,25 +1,283 @@
 # Developer Guide
 
 ## Design & implementation
+<!-- @@author iamchenjiajun -->
+### Architecture
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+##### Architecture Diagram
+Below is an architecture diagram of termiNus.
+
+![ArchitectureDiagram](./images/ArchitectureDiagram.png)
+
+- `Duke` is the main object of the program and handles all the logic and models related to the program.
+- `Ui` is the main object that provides an interface between `Duke` and the user.
+- `Command` represents a command that is provided by the user. `Ui` reads the command and it is sent to `Parser` to create a new `Command` object.
+- `Duke` executes the command and shows the output to the user using `Ui`.
+- `Command` object modifies the state of `ItemList`, which consists of multiple lists for different types of `Items`.
+- `Storage` takes the state of `ItemList` and stores it to file.
+
+##### Sequence Diagram
+Below is a sequence diagram of how the main program functions.
+
+![DukeSequenceDiagram](./images/DukeSequenceDiagram.png)
```

Comment:
Is ItemList supposed to keep existing when Duke is destroyed?
![image](https://user-images.githubusercontent.com/26886705/97539912-c35cef00-19fd-11eb-8234-dc4071dead90.png)